### DISCUSSION TITLE
Add recommended repo list

### DISCUSSION THREAD

**Discussion by bdougie** [open-sauced/open-sauced#1097] [Ideas]
- [x] User should get a list of recommend repos (probably just Javascript). 
- [ ] ~The Repos will need a structured onboarding.~ 
- [x] Need a centralized datastore to identify and share onboarding steps (thinking a separate gatsby app with a cms).

**Comment by bdougie** [comment]
It might be good to leverage what Gatsby is doing on their site for favorites. 

![Screenshot 2019-11-05 08 15 24](https://user-images.githubusercontent.com/5713670/68225003-8ace3500-ffa4-11e9-8772-2ac46f85b936.png)

**Comment by bdougie** [comment]
Update: I am going to work on making this a place to add project recommendations looking for contributions.

**Comment by bdougie** [comment]
<img width="586" alt="Screen Shot 2021-01-22 at 4 03 29 PM" src="https://user-images.githubusercontent.com/5713670/105546600-94d47480-5ccb-11eb-95d7-d1ae4c5ac7dd.png">

here is a brought idea of where it will go (in the red) above the current repo list. I am thinking of using Supabase [flava-flav](https://github.com/open-sauced/flava-flav) project to accomplish this.

**Comment by filiptronicek** [comment]
Great idea @bdougie, what do you think the source of the repos should be? Community issues? [Up for grabs](https://up-for-grabs.net/#/)?

**Comment by bdougie** [comment]
I would love to create a DB and user-generated data. I do like the approach of creating an Open Sauced GitHub App that is installed on participating repos to be recommended. We could showcase participating projects on the home page. 

These projects do something similar:

https://www.firsttimersonly.com/
https://github.com/hoodiehq/first-timers-bot

**Comment by filiptronicek** [comment]
Cool, that looks legit, I would love to see that. Would we sort the projects by any means? The number of issues? Stars?

**Comment by bdougie** [comment]
![Screen Shot 2021-04-16 at 3 59 38 PM](https://user-images.githubusercontent.com/5713670/115081504-60d7ac80-9ed2-11eb-9993-306bd21a5310.png)

[open sauced discord](https://discord.com/invite/U2peSNf23P) members had a discussion about the future of this feature recommending repos to contribute to. 

Some ideas that came up

- Leverage [orbit.love](https://orbit.love) data
- Recommend projects you have starred
- Watch stars from other (opted-in) open sauced users.

Thoughts on implementation
- run cron 4 times a day (to cover different time zones)
- only collect data from the last 7 days. 
- Encourage more contributions to projects from this.

**Comment by 0-vortex** [comment]
There are a few ways to go at this I want to suggest based on personal experience crawling my network data:

### 1. No consent way

No consent way is possible if the workflows don't store information that could be computed to expose private information and the data that is stored is also available publicly, easily. Here I would suggest the "awesome" list approach initially, where you have a graph of the users participating in open sauced and curate the result of the crons - in this case, which of the repositories your stargazers are stargazing that are also gaining traction and individually curate a gist, repository in multiple forms. This approach would produce enhanced analytics for stargazers' interests that would otherwise be unavailable or out of reach.

The tricky part here is not storing any historic information in the cron output that could be used to target different groups of users for example check new stargazers differential, which of your stargazers have exposed emails, etc.

### 2. The consent way

The consent way opens up a whole new world of possibilities I will call "garden". Here users would interact with a repository and attach themselves in different capacities to the garden based on their usage of GitHub:
- users with a lot of exposure to open source repositories can attach all their stars, watching, comments, and pull requests to the public or unfiltered layer
- users with a lot of exposure to private repositories and otherwise closed networks can generate attention to existing information and participate in the discovery process
- users with no exposure to GitHub or tech, in general, can use the available interactions to learn and potentially contribute

The tricky part here is keeping the 3 non-exclusive attention groups in a verifiable contribution layer, hindering ill-intended actors from manipulating the garden in ways that would bias the results for example by adding inconsequential data, triggering spam actions, etc.

**Comment by jasonericdavis** [comment]
The idea that I had was to include repositories that a user has starred in open sauced. My thought comes from the idea that sometimes I will star a repo for a particular reason and after a while I forget why I starred it. It would be great if I could also have that repository show up in open-sauced so that I can include notes about why I starred the repo.

**Comment by bdougie** [comment]
> The tricky part here is keeping the 3 non-exclusive attention groups in a verifiable contribution layer, hindering ill-intended actors from manipulating the garden in ways that would bias the results for example by adding inconsequential data, triggering spam actions, etc.

We should call this trifecta "the vortex." I like the idea of weighting the data based on the following, comments, etc. This would ensure the day is not following bot accounts and looking for I guess... influencers. I also think the consent should be stated at sign up, and data should be readable. 

> The idea that I had was to include repositories that a user has starred in open sauced. My thought comes from the idea that sometimes I will star a repo for a particular reason and after a while, I forget why I starred it. It would be great if I could also have that repository show up in open-sauced so that I can include notes about why I starred the repo.

Perhaps when you log in you can get a list of the last repo you starred asking to add notes on why you starred it and if you want to participate in the vortex.

**Comment by jasonericdavis** [comment]
> 
> Perhaps when you log in you can get a list of the last repo you starred asking to add notes on why you starred it and if you want to participate in the vortex.

I think that is an excellent idea. That way a user isn't bombarded with starred repos the first time.

**Comment by 0-vortex** [comment]
wtb ai powered suggestions:

<img width="173" alt="Screenshot 2021-04-17 at 00 15 05" src="https://user-images.githubusercontent.com/237133/115089200-09593100-9f12-11eb-9f1c-12b8845ed6de.png">

**Comment by bdougie** [comment]
For more style ideas. Crazy max shared this on stream. 

https://crazymax.dev/

<img width="1362" alt="Screen Shot 2021-04-20 at 5 01 14 PM" src="https://user-images.githubusercontent.com/5713670/115463719-23d52800-a1fa-11eb-9cfb-5b7ff1c39d84.png">

https://github.com/crazy-max/crazy-max.github.io/blob/dev/data/sections/projects.yaml
